jbmc/src/java_bytecode/java_trace_validation.h

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************\
+
+Module: Java trace validation
+
+Author: Jeannie Moulton
+
+\*******************************************************************/
+
+#ifndef CPROVER_JAVA_BYTECODE_JAVA_TRACE_VALIDATION_H
+#define CPROVER_JAVA_BYTECODE_JAVA_TRACE_VALIDATION_H
+
+#include <util/optional.h>
+
+class goto_tracet;
+class namespacet;
+class exprt;
+class struct_exprt;
+class symbol_exprt;
+class member_exprt;
+class messaget;
+
+/// Checks that the structure of each step of the trace matches certain
+/// criteria. If it does not, throw an error. Intended to be called by
+/// the caller of \ref build_goto_trace, for example
+/// \ref java_multi_path_symex_checkert::build_full_trace()
+void check_trace_assumptions(
+  const goto_tracet &trace,
+  const namespacet &ns,
+  const messaget &log,
+  const bool run_check = false);
+
+/// \return true iff the expression is a symbol expression and has a non-empty
+/// identifier
+bool check_symbol_structure(const exprt &expr);
+/// Recursively extracts the first operand of an expression until it reaches a
```

Comment:
⛏ A newline after the declaration would be appreciated to give it a bit more air.

Reply:
:cloud: